Replace object getter with `Object.defineProperty` in TestRenderer

flarnie opened the pull request:
**what is the change?:**
Replaces an Object Getter [1] with `Object.defineProperty`.

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get

**why make this change?:**
Our internal build system does not support object getters.

**test plan:**
We should do follow-up work to ensure that this doesn't come up again -
we can't commit code which uses object getters.

**issue:**
No issue opened yet

sebmarkbage commented:
We have tried to avoid transforms that potentially add a lot of code because it will lead to unexpectedly slow code and larger file sizes. I think this might be one right?

I'd prefer to check in the changed code.

flarnie commented:
Sure thing - will check in the changed code.

We should come up with some other way of making sure this doesn't happen again - still have not figured out how I could have checked that this syntax was not supported.

We could either add a lint or something on our side, or discover a new step for testing when syncing.

bvaughn commented:
Why would we need this transform or code-change? Getters for non-computed props are well-supported (IE9+).

bvaughn commented on src/renderers/testing/ReactTestRendererFiberEntry.js:
```diff
@@ -611,6 +605,21 @@ var ReactTestRendererFiber = {
+          return wrapFiber(root.current.child);
+        },
+        enumerable: true,
+        configurable: true,
```
I'm curious- why does this need to be enumerable or configurable?

flarnie replied:
I was copying what the babel transform would do - seems like a reasonable default. They would both default to `false` otherwise.

Since it's not `_root`, it seems like it should be enumerable and configurable.

bvaughn replied:
`enumerable` perhaps but I don't think it should be `configurable`. (Both of these default to `false` by the way, which I think would also be reasonable in this case.) 😄 Not a big deal though just sharing my thoughts

sebmarkbage replied:
Pretty much every property we have is `enumerable` and `configurable` since that's the default for assignment and object literal construction. So why should this property be different? The reason configurable is useful is that it allows for patching, polyfilling and backward compatibility fixes. The downside is that it might allow for bad patching, polyfilling etc. :)

bvaughn replied:
It felt weird for a getter that might throw to be enumerable. I don't hold this opinion strongly though. 😄

flarnie commented:
Updated things a bit because:

- Flow doesn't seem to play nicely with `Object.defineProperty/ies`     
  calls, specifically when you implement `get` and `set` methods.       
  See https://github.com/facebook/flow/issues/1336                      
- Switched from `properties` to `property` because it seems more        
  conventional in our codebase to use `property`, and I'm only setting  
  one anyway.
